### ISSUE TITLE
update_payload_cache_size crashes

### ISSUE THREAD

**Issue by bwatters-r7** [rapid7/metasploit-framework#16626]
`update_payload_cache_sizes` script fails because it is trying to compile nil.
@smcintyre-r7 is there any chance our recent changes to encrypted_shell are behind this?
I'll dig into it if it does not look familiar.

```
[ruby-3.0.2@metasploit-framework](upstream-master) tmoose@ubuntu:~/rapid7/metasploit-framework$ git log | head -n 4
commit a1613d6070cd9a481c813b8abc12c2acf9b89e3d
Author: Metasploit Bot <metasploit@rapid7.com>
Date:   Thu May 26 12:04:57 2022 -0500

[ruby-3.0.2@metasploit-framework](upstream-master) tmoose@ubuntu:~/rapid7/metasploit-framework$ ./tools/modules/update_payload_cached_sizes.rb
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
/home/tmoose/rapid7/metasploit-framework/lib/msf/core/payload/windows/encrypted_reverse_tcp.rb:193:in `get_compiled_shellcode': undefined method `compile_c' for nil:NilClass (NoMethodError)
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/payload/windows/encrypted_reverse_tcp.rb:82:in `generate'
	from /home/tmoose/rapid7/metasploit-framework/modules/payloads/adapters/cmd/windows/powershell.rb:38:in `generate'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/payload.rb:299:in `generate_complete'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/encoded_payload.rb:118:in `generate_raw'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/encoded_payload.rb:74:in `generate'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/encoded_payload.rb:24:in `create'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/base/simple/payload.rb:52:in `generate_simple'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/base/simple/payload.rb:139:in `generate_simple'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/util/payload_cached_size.rb:118:in `block in is_dynamic?'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/util/payload_cached_size.rb:117:in `map'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/util/payload_cached_size.rb:117:in `is_dynamic?'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/util/payload_cached_size.rb:127:in `is_cached_size_accurate?'
	from ./tools/modules/update_payload_cached_sizes.rb:33:in `block in <main>'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/module_set.rb:272:in `block in each_module_list'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/module_set.rb:245:in `each'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/module_set.rb:245:in `each_module_list'
	from /home/tmoose/rapid7/metasploit-framework/lib/msf/core/module_set.rb:92:in `each_module'
	from ./tools/modules/update_payload_cached_sizes.rb:29:in `<main>'

```

**Comment by smcintyre-r7** [comment]
Yeah the encrypted shells are still broken. There were two issues, one has been fixed already, the other needs #16597 to be landed before it'll be fixed. Looking at the stack trace, it looks related to the encrypted shells not generating properly.

**Comment by bwatters-r7** [comment]
Makes sense; I've updated the script locally to be less brittle on errors so it will still run and update values for successful payloads; I'll throw it up as a PR to see if it is something we'd like to bring in....
